src/subscript/ri_wellmod/ri_wellmod.py

```diff
@@ -0,0 +1,586 @@
+
+    :return: True/False
+    """
+    ecl_path = Path(ecl_case)
+    has_grid = (
+        ecl_path.with_suffix(".EGRID").exists()
+        or ecl_path.with_suffix(".GRID").exists()
+    )
+    has_init = ecl_path.with_suffix(".INIT").exists()
+    return has_grid and has_init
+
+
+def has_restart_file(ecl_case):
+    """
+    Check if ecl_case has a restart file
+    (Currently required for LGR creation, to be fixed in next ResInsight release )
+    """
+    ecl_path = Path(ecl_case)
+    return (
+        ecl_path.with_suffix(".UNRST").exists()
+        or ecl_path.with_suffix(".X0000").exists()
+    )
+
+
+def rsp_extract_export_names(well_project, well_path_names):
+    """
+    Extract export well names from ResInsight project
+
+    :param well_project: ResInsight project (rsp) file
+    :param well_path_names: List of well names (as returned by the RI instance)
+
+    :return: Dict export_name[well_path_name] = export_name
+    """
+    doc = xml.dom.minidom.parse(well_project)
+    name_nodes = doc.getElementsByTagName("WellNameForExport")
+    export_names = [node.childNodes[0].nodeValue for node in name_nodes]
+    if len(export_names) != len(well_path_names):
+        logger.error("Could not find export names for all wells - returning empty dict")
+        return {}
+
+    return dict(zip(export_names, well_path_names))
+
+
+def decode_lgr_spec(spec):
+    """
+    Decode LGR spec and return as parsed tuple. For ERT FM replace ',' by ';'.
+
+    :param spec: LGR spec in the format 'wname:ref_i,ref_j,ref_k'
+
+    :return: Tuple (str, int, int, int) if well-formed, None if not
+    """
+    tokens = re.split(",|:|;", spec.strip())
+    if len(tokens) != 4:
+        logger.warning("Malformed LGR spec string: %s", spec)
+        return None
+    wname = tokens[0]
+    try:
+        ref_i = int(tokens[1])
+        ref_j = int(tokens[2])
+        ref_k = int(tokens[3])
+    except ValueError:
+        logger.warning("Unable to integer valued refinements from LGR spec %s", spec)
+        return None
+
+    return (wname, ref_i, ref_j, ref_k)
+
+
+def split_arg_string(arg_string):
+    """
+    Split an arg string of formats 'str1,str2,str3' or 'str1|str2|str3'
+    """
+    return [tok.strip() for tok in re.split(r",|;", arg_string.strip())]
+
+
+def main():
+    """
+    Main function
+    """
+    parser = get_parser()
+    args = parser.parse_args()
+
+    debug = args.debug
+    verbose = args.verbose
+    silent = args.silent
+    well_project = args.well_project
+    ecl_case = args.ecl_case
+    input_property_files = args.property_files
+    tmp_output_folder = args.tmpfolder
+    output_file = args.output_file
+    lgr_output_file = args.lgr_output_file
+    wells = args.wells
+    msw_wells = args.msw_wells
+    lgr_specs = args.lgr
+
+    time_step = args.time_step
+    version = args.version
+
+    # Use time stamp to ensure unique output folders
+    tmp_output_folder = Path(tmp_output_folder) / str(datetime.now())
```

Comment:
`tempfile` should be safer to use than a timestamp.

Reply:
Ok, updated.